ev-rvs/nodejs-server-exercise
# feat(evobe-89): added tests and docker test container to run tests

### Ticket
https://phunts.atlassian.net/browse/EVOBE-89

### Changes Made
- List changes and reasoning in bullet points

### Notes
*Dependency [PR](https://github.com/ev-rvs/nodejs-server-exercise/pull/5) to reduce files: *

### Steps to Test
1. Checkout this branch
2. Copy `env.example` as `.env` and update `.env` file with `postgres` initialization variables
3. Purge old docker volumes with `docker volumes prune` (optional)
4. Take down existing containers by running `docker-compose down`
5. Then run test containers by running `docker-compose -f test-docker-compose.yml up`
6. Verify test were successful - see screenshot
7. Stop test containers with `docker-compose down`
4. Restart the docker-compose group `docker-compose down -v && docker-compose up --build` optionally with `-d `as a daemon or without for running logs to container.
5. Go to `http:localhost:3003/pong` and you should see JSON output

<img width="1434" alt="Screen Shot 2021-03-11 at 3 54 08 PM" src="https://user-images.githubusercontent.com/80280063/110853778-07041700-8282-11eb-90c6-94aa66198e81.png">
